### ISSUE TITLE
maven-publish:  POM file is invalid, Duplicated tag: 'dependencyManagement'

### ISSUE THREAD

**Issue by morohon** [gradle/gradle#21277]
maven-publish , when executing the generatePomFileForMavenJavaPublication task, generates an incorrect pom-default.xml containing two dependencyManagment sections, provided that the BOM file is used as a dependency:
`
implementation platform('com.amazonaws:aws-java-sdk-bom:1.12.262')
`
The problem itself and the way to solve it are described in more detail in this issue: https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/257

### Expected Behavior

maven-publish generates correct pom-default when calling artifact publish task

### Current Behavior

maven-publish throws an error when trying to publish an artifact:

```
* What went wrong:
> Execution failed for task ':publishMavenJavaPublicationToMavenLocal'.
> Failed to publish publication 'mavenJava' to repository 'mavenLocal'
> Invalid publication 'mavenJava': POM file is invalid. Check any modifications you have made to the POM file.
```

full stactrace: https://gist.github.com/morohon/516ed4670310284f36ac1a7f0dc4e78c

### Context

The problem does not allow publishing the artifact to the repository correctly without using the hack described in: https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/257#issuecomment-895790557

### Steps to Reproduce 

1. Clone repository: https://github.com/morohon/maven-publish-error
2. Execute command `./gradlew clean build generatePomFileForMavenJavaPublication`
3. Check $buildDir/publications/mavenJava/pom-default.xml - is incorrect structure.
4. Execute command `/.gradlew clean build publishMavenJavaPublicationToMavenLocal` to see error
### Your Environment

Build scan URL: https://gradle.com/s/vr4gnabdwwdm6

**Comment by jbartok** [comment]
Not a regression, tried versions as far back as 7.3.3 (using Java 17 made it not straightforward to go further back). Genuine bug, but not regression.

**Comment by xiaoyunwu** [comment]
is this fixed? Having the same issue.

**Comment by kiranshm973** [comment]
See https://github.com/jfrog/build-info/issues/198#issuecomment-1317059453

**Comment by kryspetrie** [comment]
Please, someone... fix this. The "just use the same format" thing falls apart when you're using 3rd party plugins.

**Comment by dbardbar** [comment]
This is truly annoying. Fully agree with @kryspetrie .I've implemented the hack suggested from https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/257#issuecomment-895790557, but it is horrible.

**Comment by jvandort** [comment]
This appears to be a problem with the Spring Dependency Management Plugin. See the [plugin code](https://github.com/spring-gradle-plugins/dependency-management-plugin/blob/main/src/main/java/io/spring/gradle/dependencymanagement/internal/StandardPomDependencyManagementConfigurer.java#L109-L129) which attempts to determine whether the `dependencyManagement` node already exists. 

The root of the issue is the line in `findChild`:
`if ((childObject instanceof Node) && ((Node) childObject).name().equals(name))`

The `name()` method returns an `Object`. The plugin blindly assumes this is a `String` and directly calls `.equals(name)` on it. However, `name()` in fact returns a `QName`. The plugin should compare the `localPart` of the value returned from `name()` with the `name` parameter.

**Comment by morohon** [comment]
For everyone who was interested in this problem - apparently it is fixed in version 1.1.4 dependency-management plugin: https://github.com/spring-gradle-plugins/dependency-management-plugin/issues/257 .
This is to be thanked @wilkinsona и @jvandort
